spec/lib/krikri/harvesters/oai_harvester_spec.rb

```diff
@@ -329,6 +329,21 @@
                                        .and_return(result)
             subject.send(method, { :skip_set => 'moomin' }).first
           end
+
+          it 'skips sets that error' do
+            args[:oai].delete(:set)
+            invalid = args[:oai].dup
+            invalid[:set] = 'invalid'
+            valid = args[:oai].dup
+            valid[:set] = 'valid'
+
+            allow(subject).to receive(:sets).and_return(['invalid', 'valid', 'moomin'])
+            expect(subject.client).to receive(request_type).with(invalid)
+                                       .and_raise(OAI::Exception, '')
+            expect(subject.client).to receive(request_type).with(valid)
+                                       .and_return(result)
+            subject.send(method, { :skip_set => 'moomin' }).first
+          end
```

Comment:
> Do you think it's OK that there are two expectations in this spec?

Only sort of... it's not ideal.

I thought about refactoring to use a custom matcher, but the likely shape of that matcher didn't really pop out to me when I thought about it.

Alternatively, we could conform with the letter of the law (but maybe less the spirit) by changing the first `expect` to `allow`.  That would test the skipping more precisely, but would be a bit more fragile in that it would pass if for some reason the "invalid" set was just never called.